tests/c/trace_too_long.c

```diff
@@ -0,0 +1,46 @@
+// # This test can be very slow, so ignore it except where we know it'll run
+// # fast enough.
+// ignore-if: test "$YKB_TRACER" != "hwt"
+// Run-time:
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YKD_PRINT_JITSTATE=1
+//   stderr:
+//     ...
+//     jit-state: stop-tracing
+//     jit-state: trace-compilation-aborted: Trace too long
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+int main(int argc, char **argv) {
+  YkMT *mt = yk_mt_new(NULL);
+  yk_mt_hot_threshold_set(mt, 0);
+  YkLocation loc1 = yk_location_new();
+  YkLocation loc2 = yk_location_new();
+
+  int i = 2000;
+  NOOPT_VAL(loc1);
+  NOOPT_VAL(loc2);
+  NOOPT_VAL(i);
```

Comment:
Is the duplicated `NOOPT_VAL(i)` something deviously clever, or a typo?

Reply:
Fixed in 3fafef6.